sql/src/main/java/io/crate/operation/rule/ingest/TransportDropIngestRulesForTableAction.java

```diff
@@ -0,0 +1,107 @@
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.AckedClusterStateUpdateTask;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.metadata.MetaData.Builder;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+@Singleton
+public class TransportDropIngestRulesForTableAction
+    extends TransportMasterNodeAction<DropIngestRulesForTableRequest, DropIngestRulesForTableResponse> {
+
+    private static final String ACTION_NAME = "crate/sql/drop_ingest_rules_for_table";
+
+    @Inject
+    public TransportDropIngestRulesForTableAction(Settings settings,
+                                                  TransportService transportService,
+                                                  ClusterService clusterService,
+                                                  ThreadPool threadPool,
+                                                  ActionFilters actionFilters,
+                                                  IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(settings, ACTION_NAME, transportService, clusterService, threadPool, actionFilters, indexNameExpressionResolver, DropIngestRulesForTableRequest::new);
+    }
+
+    @Override
+    protected String executor() {
+        // no need to use a thread pool, we go async right away
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected DropIngestRulesForTableResponse newResponse() {
+        return new DropIngestRulesForTableResponse();
+    }
+
+    @Override
+    protected void masterOperation(DropIngestRulesForTableRequest request, ClusterState state, ActionListener<DropIngestRulesForTableResponse> listener) throws Exception {
+        clusterService.submitStateUpdateTask("drop_ingest_rules_for_table", new AckedClusterStateUpdateTask<DropIngestRulesForTableResponse>(Priority.IMMEDIATE, request, listener) {
+
+            private long affectedRows = 0L;
+
+            @Override
+            public ClusterState execute(ClusterState currentState) throws Exception {
+                MetaData currentMetaData = currentState.metaData();
+                Builder mdBuilder = MetaData.builder(currentMetaData);
+                affectedRows = dropIngestRulesForTable(mdBuilder, request);
```

Comment:
if there are no rows affected, we can just return the currentState to prevent a new useless cluster state broadcast